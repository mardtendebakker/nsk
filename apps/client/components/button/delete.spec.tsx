import { fireEvent, render, waitFor } from '@testing-library/react';
import Delete from './delete';

describe('delete', () => {
  describe('tooltip mode', () => {
    it('matches snapshot', () => {
      const { asFragment, queryByText } = render(<Delete onClick={() => {}} tooltip />);
      expect(asFragment()).toMatchSnapshot();
      expect(queryByText('delete')).toBeNull();
    });
    it('handles click', async () => {
      const onClick = jest.fn();
      const { getByRole, getByText } = render(<Delete onClick={onClick} tooltip />);
      await waitFor(() => { fireEvent.click(getByRole('button')); });
      fireEvent.click(getByText('deleteConfirm'));

      expect(onClick).toHaveBeenCalled();
    });
    it('disables click', () => {
      const { getByRole, queryByText } = render(<Delete onClick={() => {}} disabled tooltip />);
      fireEvent.click(getByRole('button'));

      expect(queryByText('deleteConfirm')).toBeNull();
    });
  });
  describe('button mode', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(<Delete onClick={() => {}} />);
      expect(asFragment()).toMatchSnapshot();
    });
    it('handles click', async () => {
      const onClick = jest.fn();
      const { getByRole, getByText } = render(<Delete onClick={onClick} />);
      await waitFor(() => { fireEvent.click(getByRole('button')); });
      fireEvent.click(getByText('deleteConfirm'));

      expect(onClick).toHaveBeenCalled();
    });
    it('disables click', () => {
      const { getByRole, queryByText } = render(<Delete onClick={() => {}} disabled />);
      fireEvent.click(getByRole('button'));

      expect(queryByText('deleteConfirm')).toBeNull();
    });
  });
});
