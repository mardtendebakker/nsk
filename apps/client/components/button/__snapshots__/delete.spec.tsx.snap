// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete button mode matches snapshot 1`] = `
<DocumentFragment>
  <span
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1ir7th6-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      delete
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </span>
</DocumentFragment>
`;

exports[`delete tooltip mode matches snapshot 1`] = `
<DocumentFragment>
  <span
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-qecy08-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </span>
</DocumentFragment>
`;
